tests/core/pyspec/eth2spec/test/altair/sync_protocol/test_light_client_sync.py

```diff
@@ -0,0 +1,424 @@
+from typing import (Any, Dict, List)
+
+from eth_utils import encode_hex
+from eth2spec.test.context import (
+    spec_state_test_with_matching_config,
+    with_presets,
+    with_altair_and_later,
+)
+from eth2spec.test.helpers.attestations import (
+    next_slots_with_attestations,
+    state_transition_with_full_block,
+)
+from eth2spec.test.helpers.constants import MINIMAL
+from eth2spec.test.helpers.light_client import (
+    get_sync_aggregate,
+)
+from eth2spec.test.helpers.state import (
+    next_slots,
+    transition_to,
+)
+
+
+def setup_test(spec, state):
+    class LightClientSyncTest(object):
+        steps: List[Dict[str, Any]]
+        genesis_validators_root: spec.Root
+        store: spec.LightClientStore
+
+    test = LightClientSyncTest()
+    test.steps = []
+
+    yield "genesis_validators_root", "meta", "0x" + state.genesis_validators_root.hex()
+    test.genesis_validators_root = state.genesis_validators_root
+
+    next_slots(spec, state, spec.SLOTS_PER_EPOCH * 2 - 1)
+    trusted_block = state_transition_with_full_block(spec, state, True, True)
+    trusted_block_root = trusted_block.message.hash_tree_root()
+    bootstrap = spec.create_light_client_bootstrap(state)
+    yield "trusted_block_root", "meta", "0x" + trusted_block_root.hex()
+    yield "bootstrap", bootstrap
+    test.store = spec.initialize_light_client_store(trusted_block_root, bootstrap)
+
+    return test
+
+
+def finalize_test(test):
```

Comment:
Nitpicking the naming. I thought it is about finality. 😂 Maybe `finish_test` or `complete_test`?